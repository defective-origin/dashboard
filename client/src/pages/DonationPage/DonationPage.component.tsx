import React from 'react'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
import Page, { PageProps } from 'pages/Page'

// ---| screens |---
import DonationTable from 'screens/tables/DonationTable'
import PlaceholderCard from 'screens/cards/PlaceholderCard'

// ---| components |---
import Block from 'components/Block'

// ---| self |---
import css from './DonationPage.module.scss'

export type DonationPageProps = PageProps

/**
 * Component description.
 *
 * How to use
 * @example
 * <DonationPage />
 */
export function DonationPage(props: DonationPageProps): JSX.Element {
  const { children, className, ...otherProps } = props
  const _className = cn(css.DonationPage, className)

  return (
    <Page
      className={_className}
      name='PAGES.DONATION'
      menu={[{ start: 'add', tooltip: 'new request', tooltipSide: 'right', size: 'lg' }]}
      {...otherProps}
    >
      <Page.Content p='sm'>
        <Page.Section title='List of expenses' v='row' g='xs' area='1 / 1 / 3 / 4'>
          <PlaceholderCard name='Plans' />
          <PlaceholderCard name='Needs' />
          <PlaceholderCard name='Others' />

          {/* <PlaceholderCard name='History' area='3 / 1 / 4 / 4' /> */}
        </Page.Section>

        <DonationTable area='3 / 1 / 7 / 4' />

        {children}
      </Page.Content>
    </Page>
  )
}

DonationPage.displayName = 'DonationPage'

export default DonationPage
