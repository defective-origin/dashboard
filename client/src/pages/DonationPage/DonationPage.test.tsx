// ---| tests |---
import { render } from 'tests'

// ---| self |---
import DonationPage from './DonationPage.component'

describe('[DonationPage] component', () => {
  it('should render component', () => {
    const container = render(<DonationPage />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
