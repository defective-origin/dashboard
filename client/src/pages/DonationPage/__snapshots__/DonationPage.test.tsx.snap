// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[DonationPage] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="_DonationPage_49efe2"
  />
</DocumentFragment>
`;
